src/containers/ActivityInfo.jsx

```diff
@@ -10,40 +10,65 @@ const ActivityInfo = (props) => {
   const totalTeamsArr = useContext(TeamsContext);
 
   // Initial state set to current activities props with associated team information
-  // location.state is passed in from ActivityCard = 
+  // location.state is passed in from ActivityCard =
   const [currActivity, setCurrActivity] = React.useState(location.state);
   console.log('currActivity props', currActivity);
 
-
   const deleteActivity = () => {
     // Alex:Backend DELETE functionality
     console.log('Deleting activity and returning home');
     // Find current team in our context data to remove activity
     const teamsContextClone = [...totalTeamsArr];
+    let targetedAct;
     for (const team of teamsContextClone) {
       if (team.teamName === currActivity.teamName) {
-        const activityIdx = team.teamActivities.indexOf(currActivity.activity);
-        team.teamActivities.splice(activityIdx, 1);
+        for (let i = 0; i < team.teamActivities.length; i++) {
+          // loop through team activities array
+          // this method is used because the currActivity.activity object is not strictly equal to the team.teamActivities[i] object
+          // thus indexOf cannot be used here, as it will return a -1 since there are no matches in the teamActivities array
+          if (
+            // if activity in team array is equal to activity in current activity object
+            team.teamActivities[i].activity === currActivity.activity.activity
+          )
+            // remove targeted activity from team array and store it in targetedArray variable to send to server to delete
+            targetedAct = team.teamActivities.splice(i, 1);
+        }
+        console.log('targetedAct', targetedAct[0].activity);
+        console.log('teamID', team.team_id);
+        // Harvey: Send delete request to server to delete targeted card
+        fetch('/db/deleteActivity', {
+          method: 'DELETE',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            activityName: targetedAct[0].activity, // 'Go to a local thrift shop'
```

Comment:
I think test comments can be omitted in the future

Reply:
Agreed, noted!